### PR TITLE
Improved ICA based noise removal for reference channels

### PR DESCRIPTION
**Pull request by jshanna100**
Continuing  #5807, which enabled the use of ICA components derived from MEG reference channels to be used as a ground truth for selecting MEG data channel based ICA components for removal.

@larsoner had suggested in #5807 to instead do one ICA on both reference and data channels, and select those components whose weights lay most heavily on the reference channels. I decided against this simply to avoid having to write an algorithm for deciding what is heavy and what's not. However, after having applied both approaches to a larger set of data, it seems clear to me that this approach is markedly superior, and worth the effort. An additional advantage is that it is simpler for the user to implement. The procedure goes as follows:

1) Single ICA fit on data and reference channels together.
2) For each component, norm the weights in the reference channels.
3) These weight norms will have an exponential distribution, so apply log-transform to make normal.
4) Log-norm-weights can then be submitted to the iterated outlier detection already implemented in MNE.
5) Components with a z-score beyond threshold may be eliminated.

mne.preprocessing.bads.find_outliers had to be modified such that it has the option to select outliers from only one tail of the z distribution. Otherwise components could conceivably be eliminated for being especially **light** on the reference channels, which is very much undesired.

This also fixes a bug in #5807, where ICA.labels_ got reset at every new application of a find_bads procedure. Now they accumulate across multiple applications, as they should.

### CODE REVIEW

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -1208,6 +1207,9 @@ def find_bads_ref(self, inst, ch_name=None, threshold=3.0, start=None,
             Low pass frequency.
         h_freq : float
             High pass frequency.
+        method : {'together', 'separate'}
+            Method to use to identify reference channel related components.
+            Defaults to "separate." See notes.
```
Comment (agramfort) [inline mne/preprocessing/ica.py]: separate. -> separate

(no .)

**Comment by agramfort** [comment]
could we have some improvement to the documentation at the same time? with how many subject did you evaluate this? Is this approach published?

**Comment by jshanna100** [comment]
Yes, I'll work on the docs a bit.

I've now looked at over 20 recordings with it. Not published, but I do think I'll try to make short article out of it.

**Comment by agramfort** [comment]
it this not a bit premature to push this first in MNE?

**Comment by jshanna100** [comment]
Perhaps; not my decision. In any case, the bug ought to be fixed.

**Comment by agramfort** [comment]
sorry I missed something. What is the bug fix here? sorry If I did not look
close enough.

>

**Comment by jshanna100** [comment]
See last line of initial comment. Bug left over from #5807

**Comment by agramfort** [comment]
ok then can we jsut have a bug fix PR here and see for the rest later after more validation? thanks

**Pull request closed without merging**